### PR TITLE
Add support for `rails new --server falcon ...`.

### PR DESCRIPTION
**Pull request by ioquatix**
### Motivation / Background

This Pull Request has been created because using `falcon` (or indeed, any other server, e.g. `pitchfork`) should be easy.

### Detail

This Pull Request changes `rails new` and adds support for `--server puma` or `--server falcon`.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: railties/lib/rails/generators/app_base.rb [hunk 1]
```diff
@@ -281,7 +287,11 @@ def database_gemfile_entry # :doc:
       end
 
       def web_server_gemfile_entry # :doc:
-        GemfileEntry.new "puma", ">= 5.0", "Use the Puma web server [https://github.com/puma/puma]"
+        if options[:server] == "falcon"
+          GemfileEntry.new "falcon", ">= 0.28.0", "Use the Falcon web server [https://github.com/socketry/falcon]"
```
Comment (santib) [inline railties/lib/rails/generators/app_base.rb:291]: just curious, any reason why not to set the `>=` to the latest version?

**Comment by rafaelfranca** [comment]
Thank you for the pull request, but I don't think we want to hardcode more things in the Rails generator. Changing from puma to falcon is just a matter of changing the gemfile and adding a line in application.rb. It isn't worth an option.

**Comment by ioquatix** [comment]
@rafaelfranca 

> I don't think we want to hardcode more things in the Rails generator

Why is that a problem?

> Changing from puma to falcon is just a matter of changing the gemfile and adding a line in application.rb. It isn't worth an option.

Making it easier for users is a good idea, no? It's definitely easier to explain `rails new --server falcon` vs "change several files".

I'd like to make it easier for users.

**Comment by eugeneius** [comment]
Also from the linked PR:

> I'm wondering if this should be turned on automatically by falcon using a Railtie?

This seems like a better option, since it will make Falcon work correctly even if a user adds it to their Gemfile manually.

**Comment by rafaelfranca** [comment]
One problem is that we don't want to be changing Rails every time a new webserver, or database wants to be added here. A better approach would be to add a install command on falcon so you can add the necessary file you need to make falcon to work. Or you can take the approach @eugeneius suggested.

The other problem is that this is basically endorsing falcon, which we don't want to do. Nobody in the core team uses Falcon, so I don't think this is something we can/should endorce.

Just want to point out that we take the same positions with a lot of things. We don't have a `rails new foo --test-framework=rspec`, and we are going to remove all database adapters that aren't `mysql`, `sqlite` and `postgres`.

**Comment by ioquatix** [comment]
> This seems like a better option, since it will make Falcon work correctly even if a user adds it to their Gemfile manually.

Yeah, I agree with this and it looks like a great idea, but it's still a separate issue from this PR. I'll need to investigate what's possible.

**Comment by ioquatix** [comment]
For reference, we did add a Railtie to Falcon so it does the correct thing by default.

**Pull request closed without merging**